types/iron/tslint.json

```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```

Comment:
Why disable all these rules?

Reply:
Commit 82c407a. I enabled some rules. But I am not sure if it's ok. What rules should be enable in this case? Thanks!

Reply:
Interesting, `strict` doesn't seem to work for ts 2.4, I think it would be safe to up it to 2.6.

Reply:
Ideally we should stick to the suggest base rules without any changes.

Reply:
@SimonSchick In my local environment I changed the "index.d.ts" to use TS 2.6:

```
// TypeScript Version: 2.6
```

And I executed "npm run lint iron" to test it, but is the same problem unfortunately:

```
$ tsc -v
Version 2.6.2

$ npm run lint iron
Error: Unexpected compiler option strict
    at Object.<anonymous> (/mnt/d/Workspace/DefinitelyTyped-Rafael/node_modules/dtslint/bin/checks.js:81:35)
    at Generator.next (<anonymous>)
    at fulfilled (/mnt/d/Workspace/DefinitelyTyped-Rafael/node_modules/dtslint/bin/checks.js:4:58)
    at <anonymous>
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! definitely-typed@0.0.1 lint: `dtslint types "iron"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the definitely-typed@0.0.1 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

If I rollback the changes, it works fine. I tried to figure out why it happen but I don't have any answer yet. I am looking for another types definitions and apparently no one is using it.

Should I revert the commit?

Reply:
I believe DT automatically installs the needed versions here, maybe just push the code and see what happens lol.

Reply:
Ohhh, the same :-/

```
Error in iron

Command failed: node /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/index.js 

Error: Unexpected compiler option strict

    at Object.<anonymous> (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/checks.js:81:35)

    at Generator.next (<anonymous>)

    at fulfilled (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/checks.js:4:58)

    at <anonymous>
```

Reply:
I saw, not really sure what the issue is here....

Reply:
Oh, it seems strict is _strictly_ not allowed, you need to specify all options manually instead...

Reply:
https://github.com/Microsoft/dtslint/blob/dffcc5d6877fdec7380cd48a2e4e9357a3d786f9/src/checks.ts#L78

Reply:
Sorry for the noob question, but where can I get the tslint "suggest base rules"? I am looking here and each typings has its own implementation.
Ops. It's late here man lol. I need to sleep.

Reply:
You can find them in the same repo I just linked, ideally your tslint should just be this:

```{ "extends": "dtslint/dt.json" }```